src/pextlib1.0/Pextlib.c

```diff
@@ -1133,6 +1134,9 @@ int Pextlib_Init(Tcl_Interp *interp)
 
     Tcl_CreateObjCommand(interp, "fs_case_sensitive", FSCaseSensitiveCmd, NULL, NULL);
 
+    Tcl_CreateObjCommand(interp, "devicenode", DeviceNodeCmd, NULL, NULL);
+    Tcl_CreateObjCommand(interp, "fs_type", FilesystemTypeCmd, NULL, NULL);
+
```

Comment:
To keep the interface closer to the C API, I would do the same as in our implementation of [getpwuid](https://github.com/macports/macports-base/blob/master/src/pextlib1.0/uid.c#L203-L306) and either return a list of key/value pairs for the `struct statfs` members or a specific field that was requested. These two examples would then give equivalent output:

```
array set data [statfs $path]
puts $data(mntfromname)
puts $data(fs_type)
```
```
puts [statfs $path mntfromname]
puts [statfs $path fs_type]
```

Although we do not need the other entries right now, we might later need them. I think it is better to keep the abstraction to the C API as low as possible here to make it easier to understand what this is meant to retrieve.

I am sorry that I did not outline this directly that clearly when I suggested a wrapper for `statfs(2)`. If you want me to, I could also look into implementing `statfs` in the same style as `getpwuid`.

Reply:
I've never written these before, so I went for low complexity so I could get it done when I had a few moments last night. If you're up for putting together a lower level version, I'm all for it!